nncf/experimental/tensorflow/quantization/quantizers.py

```diff
@@ -0,0 +1,310 @@
+        size *= input_shape[axis]
+    return size
+
+
+class QuantizerV2(Quantizer):
+    """
+    Base class for all quantization operations.
+    """
+
+    def __init__(self,
+                 name: str,
+                 qspec: TFQuantizerSpec,
+                 input_type: str,
+                 input_shape: Optional[List[int]] = None,
+                 channel_axes: Optional[List[int]] = None):
+        """
+        Initializes the internal state of the quantizer.
+
+        :param name: Name of operation. Unique identifier inside
+            the NNCF network.
+        :param qspec: Specification of the quantizer. Is a collection
+            of parameters that influence how quantization performs.
+        :param input_type: Indicates the type of input tensor: `inputs` or `weights`.
+        :param input_shape: Shape of the input tensor for which the
+            quantization is applied. Required only for per-channel
+            quantization.
+        :param channel_axes: Axes numbers of the input tensor which
+            correspond to its channels. Required only for per-channel
+            quantization.
+        """
+        super().__init__(name, qspec)
+
+        # Specification of the input tensor
+        self.input_type = input_type
+        self._input_shape = input_shape
+        self.channel_axes = channel_axes
+        if self.per_channel and (self._input_shape is None or self.channel_axes is None):
+            raise ValueError('The `input_shape` and `channel_axes` arguments are required when'
+                             'using per-channel quantization.')
+
+    @property
+    def input_shape(self):
+        return self._input_shape
+
+    def call(self, inputs, *args, **kwargs):
+        """
+        Applies quantization to the input tensor if the quantizer is enabled.
+        Otherwise, if the quantizer is disabled, returns the input tensor as-is.
+
+        :param inputs: Input tensor.
+        :return: Output tensor.
+        """
+        if not self.enabled:
+            return inputs
+        transformed = self._pre_processing_fn(inputs)
+        quantized = self.quantize(transformed)
+        outputs = self._post_processing_fn(quantized)
+        return outputs
+
+    def quantize(self, inputs):
+        """
+        Applies quantization operation to the input tensor.
+
+        :param inputs: Input tensor.
+        :return: Quantized tensor.
+        """
+        raise NotImplementedError
+
+    def apply_range_initialization(self, min_values, max_values, min_range=0.1, eps=0.01):
+        """
+        Initialize quantizer parameters using minimum and maximum weight values.
+
+        :param min_values: Minimum weight values.
+        :param max_values: Maximum weight values.
+        :param min_range: Minimum range.
+        :param eps: Smoothing coefficient for ranges: min_range = maximum(min_range, eps * max_range).
+        """
+        raise NotImplementedError
+
+    def setup_input_transformation(self):
```

Comment:
Docstring missing

Reply:
Thanks for the comment! I added the missing docstring.